<a href="https://colab.research.google.com/github/johncduran/datasci_6_anova/blob/main/HHA_507_Assignment_6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Importing

In [17]:

import pandas as pd
import numpy as np
import scipy.stats as stats
from scipy.stats import shapiro, levene, f_oneway


In [3]:
df = pd.read_csv('https://raw.githubusercontent.com/johncduran/datasci_6_anova/main/datasets/diabetic_data.csv')
df



encounter_id  patient_nbr             race  gender      age weight  \
0            2278392      8222157        Caucasian  Female   [0-10)      ?   
1             149190     55629189        Caucasian  Female  [10-20)      ?   
2              64410     86047875  AfricanAmerican  Female  [20-30)      ?   
3             500364     82442376        Caucasian    Male  [30-40)      ?   
4              16680     42519267        Caucasian    Male  [40-50)      ?   
...              ...          ...              ...     ...      ...    ...   
101761     443847548    100162476  AfricanAmerican    Male  [70-80)      ?   
101762     443847782     74694222  AfricanAmerican  Female  [80-90)      ?   
101763     443854148     41088789        Caucasian    Male  [70-80)      ?   
101764     443857166     31693671        Caucasian  Female  [80-90)      ?   
101765     443867222    175429310        Caucasian    Male  [70-80)      ?   

        admission_type_id  discharge_disposition_id  admission_source_id  \
0                       6                        25                    1   
1                       1                         1                    7   
2                       1                         1                    7   
3                       1                         1                    7   
4                       1                         1                    7   
...                   ...                       ...                  ...   
101761                  1                         3                    7   
101762                  1                         4                    5   
101763                  1                         1                    7   
101764                  2                         3                    7   
101765                  1                         1                    7   

        time_in_hospital  ... citoglipton insulin  glyburide-metformin  \
0                      1  ...          No      No                   No   
1                      3  ...          No      Up                   No   
2                      2  ...          No      No                   No   
3                      2  ...          No      Up                   No   
4                      1  ...          No  Steady                   No   
...                  ...  ...         ...     ...                  ...   
101761                 3  ...          No    Down                   No   
101762                 5  ...          No  Steady                   No   
101763                 1  ...          No    Down                   No   
101764                10  ...          No      Up                   No   
101765                 6  ...          No      No                   No   

        glipizide-metformin  glimepiride-pioglitazone  \
0                        No                        No   
1                        No                        No   
2                        No                        No   
3                        No                        No   
4                        No                        No   
...                     ...                       ...   
101761                   No                        No   
101762                   No                        No   
101763                   No                        No   
101764                   No                        No   
101765                   No                        No   

        metformin-rosiglitazone  metformin-pioglitazone  change diabetesMed  \
0                            No                      No      No          No   
1                            No                      No      Ch         Yes   
2                            No                      No      No         Yes   
3                            No                      No      Ch         Yes   
4                            No                      No      Ch         Yes   
...                         ...                     ...     ...         ...   
101761                       No

This dataset contains information of patients with diabetes from 130 plus hospitals in the U.S. from 1999-2008

## Research question:
Does one gender get more lab procedures done than another?

## Hypothesis

The null hypothesis would be that there are no significant differences between the means of male and female groups in terms of number of lab procedures.

Alternative hypothesis: There is a significant difference between the means of male and female groups in terms of number of lab procedures.



In [4]:
columns_to_keep = ['gender', 'num_lab_procedures']
df = df[columns_to_keep]

Here I am creating a new dataframe using only the two columns that I need which are 'gender' and 'num_lab_procedures'.

In [5]:

female_data = df[df['gender'] == 'Female']['num_lab_procedures']
male_data = df[df['gender'] == 'Male']['num_lab_procedures']

Here I am creating two subsets of data

In [9]:
stat_female, p_value_female = shapiro(female_data)
stat_male, p_value_male = shapiro(male_data)

/usr/local/lib/python3.10/dist-packages/scipy/stats/_morestats.py:1882: UserWarning: p-value may not be accurate for N > 5000.
  warnings.warn("p-value may not be accurate for N > 5000.")


In [13]:
print(f"Result of the Shapiro-Wilk Test (Female): Statistic={stat_female}, p-value={p_value_female}")
print(f"Result of the Shapiro-Wilk Test (Male): Statistic={stat_male}, p-value={p_value_male}")

Result of the Shapiro-Wilk Test (Female): Statistic=0.9830589294433594, p-value=0.0
Result of the Shapiro-Wilk Test (Male): Statistic=0.9865784049034119, p-value=0.0


Using the Shapiro-Wilk test both the p-values are both 0.0 which is less than the significance level of 0.05. This means that the data for both males and females is not normally distributed.

In [16]:
stat_levene, p_value_levene = levene(female_data, male_data)

print(f"Result of the Levene Test: Statistic={stat_levene}, p-value={p_value_levene}")

Levene's Test: Statistic=7.103994574748455, p-value=0.007692449230885389


Given the results of the Levene test where the p-value is 0.0, it can be concluded that the variances in the number of lab procedures are not equal in the male and female groups.

In [19]:
stat_anova, p_value_anova = f_oneway(female_data, male_data)

print(f"Result of the One-way ANOVA: Statistic={stat_anova}, p-value={p_value_anova}")

Result of the One-way ANOVA: Statistic=0.9537747007512334, p-value=0.328762658830843


Since the p-value in the ANOVA test is 0.33 which is greater than 0.05 and so we would not reject the null hypothesis. This means that there isn't enough evidence to say that there are significant differences in the means of the number of lab procedures in the male and female groups.